### PR TITLE
Fix parse errors animalqtldb flybase mmrrc

### PR DESCRIPTION
**Pull request by justaddcoffee**
This PR fixes problems causing AnimaQTLdb, Flybase, and MMRRC ingests to fail.

* AnimalQTLdb:
   * added column info to be consistent with how other ingests do row checking while ingesting files
   * fixed ingest to recover when bad data is encountered in _process_qtls_genetic_location() - two records from source contain extra columns

* Flybase:
    * fixed error while parsing fbrf_pmid_pmcid_doi_fb_.*.tsv.gz, caused by encoding not being set correctly (latin-1)

* MMRRC:
fixed two problems in parsing mmrrc_catalog_data.csv:
   * one record from source contains extra columns, causing ingest to fail because of ValueError
   * header was being parsed incorrectly, causing first two records to be silently ignored, because of change in format of header upstream at MMRRC

### CODE REVIEW

**Comment by justaddcoffee** [comment]
@TomConlin per your suggestion, I've made this PR from commits 3220828 and 69b2b79 from PR #809. I've already made some fixed to the MMRRC ingest, which was also failing.

**Comment by TomConlin** [comment]
getting there.

first will draw your attention to the conventions of `src_key` 
and `col = self.files[sec_key]`

```
ag -l "col = self.files\[src_key\]" dipper/sources/
dipper/sources/HPOAnnotations.py
dipper/sources/FlyBase.py
dipper/sources/NCBIGene.py
dipper/sources/Panther.py
dipper/sources/GeneOntology.py
dipper/sources/Coriell.py
dipper/sources/OMIM.py
dipper/sources/Monarch.py
dipper/sources/GWASCatalog.py
```

(there will be others with slight variations).

next to the explicit non-positional mapping of headers to variables 
including commenting out (but leaving in place) of unused items

```
ag -l " = row\[col.index\("  dipper/sources/
dipper/sources/Ensembl.py
dipper/sources/HPOAnnotations.py
dipper/sources/EBIGene2Phen.py
dipper/sources/Monochrom.py
dipper/sources/UDP.py
dipper/sources/GeneReviews.py
dipper/sources/FlyBase.py
dipper/sources/UCSCBands.py
dipper/sources/MMRRC.py
dipper/sources/HGNC.py
dipper/sources/NCBIGene.py
dipper/sources/GeneOntology.py
dipper/sources/Panther.py
dipper/sources/MGI.py
dipper/sources/Coriell.py
dipper/sources/EOM.py
dipper/sources/OMIM.py
dipper/sources/IMPC.py
dipper/sources/Monarch.py
dipper/sources/OMIMSource.py
dipper/sources/StringDB.py
dipper/sources/GWASCatalog.py
```

It need not be too onerous if you generate it

```
sed "s/\(.*\)/\t# \1 = row[col.index('\1')].strip()/g" whateverheader 
	# qtl_id = row[col.index('qtl_id')].strip()
	# qtl_symbol = row[col.index('qtl_symbol')].strip()
	# trait_name = row[col.index('trait_name')].strip()
	# assotype = row[col.index('assotype')].strip()
	# empty = row[col.index('empty')].strip()
	# chromosome = row[col.index('chromosome')].strip()
	# position_cm = row[col.index('position_cm')].strip()
	# range_cm = row[col.index('range_cm')].strip()
	# flankmark_a2 = row[col.index('flankmark_a2')].strip()
	# flankmark_a1 = row[col.index('flankmark_a1')].strip()
	# peak_mark = row[col.index('peak_mark')].strip()
	# flankmark_b1 = row[col.index('flankmark_b1')].strip()
	# flankmark_b2 = row[col.index('flankmark_b2')].strip()
	# exp_id = row[col.index('exp_id')].strip()
	# model_id = row[col.index('model_id')].strip()
	# test_base = row[col.index('test_base')].strip()
	# sig_level = row[col.index('sig_level')].strip()
	# lod_score = row[col.index('lod_score')].strip()
	# ls_mean = row[col.index('ls_mean')].strip()
	# p_values = row[col.index('p_values')].strip()
	# f_statistics = row[col.index('f_statistics')].strip()
	# variance = row[col.index('variance')].strip()
	# bayes_value = row[col.index('bayes_value')].strip()
	# likelihood_ratio = row[col.index('likelihood_ratio')].strip()
	# trait_id dom_effect = row[col.index('trait_id dom_effect')].strip()
	# add_effect = row[col.index('add_effect')].strip()
	# pubmed_id = row[col.index('pubmed_id')].strip()
	# gene_id = row[col.index('gene_id')].strip()
	# gene_id_src = row[col.index('gene_id_src')].strip()
	# gene_id_type = row[col.index('gene_id_type')].strip()
	# empty2 = row[col.index('empty2')].strip()
```

then uncomment any that are actually used.

not that it is necessarily relevant to this pr 
but I  will also note attempting to get the contents into the shape we will 
be using use them during assignment as much as possible. 
so any casting of types, or case switching, trimming etc
all happen in the same load block before the rest of the processing

**Comment by TomConlin** [comment]
also I don't know why String is now failing CI

**Comment by justaddcoffee** [comment]
> also I don't know why String is now failing CI

Not sure either. I restarted the test and it's passing now, without any change to the code. That test seems a little brittle at times.

**Comment by justaddcoffee** [comment]
@TomConlin okay, AnimalQTLdb should be as you describe now

File: dipper/sources/AnimalQTLdb.py [hunk 1]
```diff
@@ -599,14 +614,15 @@ def _process_qtls_genetic_location(
 
                     assoc.add_association_to_graph()
 
+                # off by one - the following actually gives us (limit + 1) records
                 if not self.test_mode and limit is not None and line_counter > limit:
```
Comment (TomConlin) [inline dipper/sources/AnimalQTLdb.py]: note: 
the non-generic readers   (e.g. filereader = csv.reader) often have builtin line counters
so 
```            filereader.line_num > limit
```

Comment (justaddcoffee) [reply dipper/sources/AnimalQTLdb.py]: Yep, this one has a built-in line counter. I'm not fixing this here in the interest of a more focused PR.

**Comment by TomConlin** [comment]
+1

**Pull request merged**